### PR TITLE
Fix SetupJemalloc with CMake 3.12

### PR DESCRIPTION
**Pull request by nilsdeppe**
### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `major new feature` if appropriate.

### CODE REVIEW

**Review by nilsvu — DISMISSED** [review]

**Review by geoffrey4444 — DISMISSED** [review]
Works with cmake 3.13.1 on ocean

File: cmake/SetupJemalloc.cmake [hunk 1]
```diff
@@ -29,33 +29,20 @@ get_filename_component(
...
-    ${JEMALLOC_LIB_NAME}
-    LAST_EXT
-    )
-endwhile()
-
-if("${JEMALLOC_LIB_TYPE}" STREQUAL ".a")
+if(APPLE)
+  string(FIND "${JEMALLOC_LIB_NAME}" ".dylib" FOUND_SHARED)
+else()
+  string(FIND "${JEMALLOC_LIB_NAME}" ".so" FOUND_SHARED)
```
Comment (nilsvu) [inline cmake/SetupJemalloc.cmake:39]: Can you use `string(REGEX MATCH ... ".so$" ...)` instead to make sure this is the last extension? If this is more complicated for some reason, please add a code comment.

**Review by nilsvu — CHANGES_REQUESTED** [review]

**Comment by nilsvu** [comment]
@nilsdeppe since Ocean now runs cmake 3.18, should we just drop CMake 3.12 and go a few versions up? Let me know which version you prefer. I would suggest 3.15 or 3.16.

**Comment by nilsdeppe** [comment]
Okay, added a comment explaining why we need a find

**Review by nilsvu — APPROVED** [review]

**Review by kidder — APPROVED** [review]

**Pull request merged**